fdbserver/MockS3Server.actor.cpp

```diff
@@ -0,0 +1,958 @@
+	static MockS3ServerImpl instance;
+	return instance;
+}
+
+// Clear singleton state for clean test runs
+static void clearSingletonState() {
+	MockS3ServerImpl& instance = getSingletonInstance();
+	instance.buckets.clear();
+	instance.multipartUploads.clear();
+	TraceEvent("MockS3ServerImpl_StateCleared");
+}
+
+// Request Handler Implementation - Uses singleton to preserve state
+Future<Void> MockS3RequestHandler::handleRequest(Reference<HTTP::IncomingRequest> req,
+                                                 Reference<HTTP::OutgoingResponse> response) {
+	TraceEvent("MockS3RequestHandler_GetInstance").detail("Method", req->verb).detail("Resource", req->resource);
+
+	// Use singleton instance to maintain state across requests while avoiding reference counting
+	MockS3ServerImpl& serverInstance = getSingletonInstance();
+
+	TraceEvent("MockS3RequestHandler_UsingInstance")
+	    .detail("InstancePtr", format("%p", &serverInstance))
+	    .detail("Method", req->verb)
+	    .detail("Resource", req->resource);
+
+	TraceEvent("MockS3RequestHandler").detail("Method", req->verb).detail("Resource", req->resource);
+
+	return MockS3ServerImpl::handleRequest(&serverInstance, req, response);
+}
+
+Reference<HTTP::IRequestHandler> MockS3RequestHandler::clone() {
+	return makeReference<MockS3RequestHandler>();
+}
+
+// Safe server registration that prevents conflicts
+ACTOR Future<Void> registerMockS3Server(std::string ip, std::string port) {
+	state std::string serverKey = ip + ":" + port;
+
+	// DIAGNOSTIC: Enhanced registration logging
+	TraceEvent("MockS3ServerDiagnostic")
+	    .detail("Phase", "Registration Start")
+	    .detail("IP", ip)
+	    .detail("Port", port)
+	    .detail("ServerKey", serverKey)
+	    .detail("IsSimulated", g_network->isSimulated())
+	    .detail("AlreadyRegistered", registeredServers.count(serverKey) > 0);
+
+	// Check if server is already registered
+	if (registeredServers.count(serverKey)) {
+		TraceEvent(SevWarn, "MockS3ServerAlreadyRegistered").detail("Address", serverKey);
+		return Void();
+	}
+
+	try {
+		TraceEvent("MockS3ServerDiagnostic")
+		    .detail("Phase", "Calling registerSimHTTPServer")
+		    .detail("Address", serverKey);
+
+		wait(g_simulator->registerSimHTTPServer(ip, port, makeReference<MockS3RequestHandler>()));
+		registeredServers[serverKey] = true;
+
+		TraceEvent("MockS3ServerRegistered").detail("Address", serverKey).detail("Success", true);
+
+		TraceEvent("MockS3ServerDiagnostic")
+		    .detail("Phase", "Registration Complete")
+		    .detail("Address", serverKey)
+		    .detail("TotalRegistered", registeredServers.size());
+	} catch (Error& e) {
+		TraceEvent(SevError, "MockS3ServerRegistrationFailed")
+		    .error(e)
+		    .detail("Address", serverKey)
+		    .detail("ErrorCode", e.code())
+		    .detail("ErrorName", e.name());
+		throw;
+	}
+
+	return Void();
+}
+
+// Public Interface Implementation
+ACTOR Future<Void> startMockS3Server(NetworkAddress listenAddress) {
```

Comment:
When is this startMockS3Server used? Thanks1

Reply:
Not currently. I thought I would need it until tripped over the register http server thingy. Seems useful though if someone wanted to start up a mock s3 instance.